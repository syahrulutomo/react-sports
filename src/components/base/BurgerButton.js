import React from 'react';
import PropTypes from 'prop-types';

export const BurgerButton = (props) => {
  const { active, handleClick } = props;

  const classStyle = active ? 'burger-button-active' : 'burger-button'; 

  return (
    <div style={{ width: '40px', height: '40px', position: 'relative', top: '50%' }} onClick={handleClick}>
      <div className={classStyle}/>
    </div>
  );
}

BurgerButton.propTypes = {
  active: PropTypes.bool,
  handleClick: PropTypes.func
}