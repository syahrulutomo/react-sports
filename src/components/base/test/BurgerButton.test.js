import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BurgerButton } from '../BurgerButton';

const renderer = new ShallowRenderer();

describe('<BurgerButton />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<BurgerButton />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
