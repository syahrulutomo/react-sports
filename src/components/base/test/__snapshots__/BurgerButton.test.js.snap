// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerButton /> should render and match the snapshot 1`] = `
<div
  style={
    Object {
      "height": "40px",
      "position": "relative",
      "top": "50%",
      "width": "40px",
    }
  }
>
  <div
    className="burger-button"
  />
</div>
`;
